src/fides/api/models/privacy_request.py

```diff
@@ -1223,6 +1231,45 @@ def verify_identity(
         self.identity_verified_at = datetime.utcnow()
         self.save(db)
 
+    def persist_custom_privacy_request_fields(
+        self,
+        db: Session,
+        custom_privacy_request_fields: Optional[
+            Dict[str, CustomPrivacyRequestFieldSchema]
+        ],
+    ) -> None:
+        if not custom_privacy_request_fields:
+            return
+
+        if CONFIG.execution.allow_custom_privacy_request_field_collection:
+            for key, item in custom_privacy_request_fields.items():
+                if item.value:
+                    hashed_value = CustomPrivacyRequestField.hash_value(item.value)
+                    CustomPrivacyRequestField.create(
+                        db=db,
+                        data={
+                            "consent_request_id": self.id,
+                            "field_name": key,
+                            "field_label": item.label,
+                            "encrypted_value": {"value": item.value},
+                            "hashed_value": hashed_value,
+                        },
```

Comment:
I added the custom fields to the `ConsentRequest` to minimize changes to the front-end. The front-end passes along the custom fields during the initial call to create a consent request but not for the subsequent privacy preference calls. Additionally, as you pointed out. We don't always create a `PrivacyRequest` for privacy preferences so we would need a separate location to store any custom fields.

I wasn't sure of the best way to handle this in the database, I don't like how there's two sets of custom fields. I was thinking of just moving the relationships between `ConsentRequest`, `PrivacyRequest`, and `CustomPrivacyRequestFields` to linking tables instead

`consent_request` <-> `consent_request_custom_fields` <-> `custom_privacy_request_fields`
`privacy_request` <-> `privacy_request_custom_fields` <-> `custom_privacy_request_fields`

But I wasn't sure if this was overkill. Open to hear what you think

Reply:
I still don't quite follow why we need custom privacy request field data if a privacy request isn't created because we're not using it for anything then? 

but assuming we do need it.  Can we link the same custom_privacy_request_field record to the consent_request_id first, and then if we come in and create a privacy request, edit those exisitng records with the privacy_request_id?

Reply:
@galvana can you look at this comment before merging? Let's make sure we're on the same page here

Reply:
In the original requirements for the custom privacy request fields work, we separated out the concepts of collecting custom fields, and using custom fields in a privacy request. In this case, if all we do is collect custom fields as part of a consent request, that is still a valid use case. With that in mind, I made the refactor you recommended above to only create one set of custom privacy request fields per consent/privacy request pair.

Reply:
OK got it thanks for clarifying that collecting custom fields is still a separate use case!